src/jit/importer.cpp

```diff
@@ -11667,6 +11667,13 @@ MATH_MAYBE_CALL_NO_OVF:         ovfl = false;
             }
 
             eeGetFieldInfo(&resolvedToken, (CORINFO_ACCESS_FLAGS)aflags, &fieldInfo);
+
+            // Check if LDSFLD tries to load a non-static field
+            if (opcode == CEE_LDSFLD && !(fieldInfo.fieldFlags & CORINFO_FLG_FIELD_STATIC))
```

Comment:
cc: There should be parentheses around all unary and binary expressions if they are contained within other expressions.